lecture07/springboot-backend/src/main/java/net/javaguides/springboot/controller/EmployeeController.java

```diff
@@ -0,0 +1,103 @@
+package net.javaguides.springboot.controller;
+
+import jakarta.validation.Valid;
+import net.javaguides.springboot.exception.ResourceNotFoundException;
+import net.javaguides.springboot.model.Employee;
+import net.javaguides.springboot.repository.EmployeeRepository;
+import net.javaguides.springboot.service.EmployeeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Rest API定義クラス
+ */
+@RestController
+@RequestMapping("/api/v1/employees")
+public class EmployeeController {
+
+    /** Employeeアクセス用リポジトリ */
+    @Autowired
+    private EmployeeRepository employeeRepository;
+    /** EmployeeService */
+    @Autowired
+    private EmployeeService service;
+
+    /**
+     * Employeeデータリストを取得する.
+     * @return Employeeデータリスト
+     */
+    @GetMapping
+    public List<Employee> getAllEmployees() {
+        return employeeRepository.findAll();
+    }
+
+    /**
+     * 指定したIDをもつEmployeeデータを取得する.
+     * @param id ID
+     * @return 指定したIDをもつユーザーデータ
+     */
+    @GetMapping("{id}")
+    public Employee getEmployeeById(@PathVariable long id){
+        Optional<Employee> employeeId = service.getEmployeeId(id);
+        if(employeeId.isPresent()) {
+            // 指定したIDをもつユーザーデータがあればそのユーザーデータを返す
+            return employeeId.get();
+        }else{
+            // 指定したIDをもつユーザーデータがなければnullを返す
+            return employeeId.orElseThrow(() -> new ResourceNotFoundException("Employee not exist width id:" + id));
+        }
```

Comment:
よくよく見ると、この処理は
```suggestion
return employeeId.orElseThrow(() -> new ResourceNotFoundException("Employee not exist width id:" + id));
```
だけでよいですね。